### PR TITLE
Create action "ADD project" for redux #12

### PR DESCRIPTION
**Pull request by tomaszprc**

### CODE REVIEW

File: src/styles/components/_button.scss [hunk 1]
```diff
@@ -0,0 +1,25 @@
+.button {
+  padding: 15px 20px;
+  border-radius: 8px;
+  background-color: #3c652e;
+  font-weight: bold;
+  text-decoration: none;
+  color: white;
+  cursor: pointer;
```
Comment (korczas) [inline src/styles/components/_button.scss:8]: let's go with asc sorting ;)

File: src/styles/components/_button.scss [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    background-color: #e93c30;
+  }
+
+  &.edit {
+    background-color: #1f5aa6;
+  }
+
+  &.add {
+    background-color: #6f7362;
+  }
```
Comment (korczas) [inline src/styles/components/_button.scss:20]: What do you think about using more generic class names, like: primary, secondary, tertiary, success, fail, etc.

**Review by korczas — CHANGES_REQUESTED** [review]
update CSS props sorting.

File: src/components/Button/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  modificator?: string;
+  large?: boolean;
+}
+
+const Button = ({ text, modificator, large }: ButtonProps) => {
+  return (
+    <div
+      className={`button ${modificator ? modificator : ""} ${
+        large ? "large" : ""
+      }`}
```
[the remark this answers is not in the document]
Comment (tomaszprc) [reply src/components/Button/Button.tsx:13]: Moved to another jira. 
https://github.com/tomaszprc/timelog_selleo/issues/20

File: src/redux/projects/reducers.ts [hunk 4]
```diff
@@ -17,10 +18,14 @@ export function projectReducer(
 ): ProjectsList {
   switch (action.type) {
     case ADD_PROJECT:
-      return {
-        projectsList: [...state.projectsList, action.payload],
+      const { projectsList } = state;
+      const id = projectsList[projectsList.length - 1].id + 1;
```
Comment (korczas) [inline src/redux/projects/reducers.ts:22]: `const id = projectsList[projectsList.length - 1]?.id + 1 || 0;`

File: src/redux/projects/types.ts [hunk 5]
```diff
@@ -7,17 +8,26 @@ export interface Project {
   timeTrackerIds: number[];
 }
 
+export type AddProjectPayload = Omit<Project, "id" | "timeTrackerIds">;
```
Comment (korczas) [inline src/redux/projects/types.ts:11]: I know I wrote this, but in this case is better to use 
```
export type AddProjectPayload = Pick<Project, "title" | "description">;
```

File: src/redux/projects/types.ts [hunk 6]
```diff
@@ -7,17 +8,26 @@ export interface Project {
...
+
 export interface ProjectsList {
   projectsList: Project[];
+  taskCount: number;
+}
+
+export interface ProjectModalState {
+  projects: ProjectsList;
+  openModal: boolean;
```
Comment (korczas) [inline src/redux/projects/types.ts:20]: not used anywhere

File: src/redux/projects/actions.ts [hunk 7]
```diff
@@ -1,6 +1,7 @@
 import { ADD_PROJECT, Project, ProjectActionTypes } from "./types";
 
 export function addProject(newProject: Project): ProjectActionTypes {
+  // request do bazy => baza zwraca async payload => payload zwracasz w return
```
[the remark this answers is not in the document]
Comment (tomaszprc) [reply src/redux/projects/actions.ts:4]: I leave it for my note, it can be helpful in future when I work with DB.

**Pull request merged**